docs/manual/faq.tex

```diff
@@ -1416,6 +1373,68 @@
 \end{itemize}
 
 
+\label{declaration-annotations-moved} % 2021-04-16 temporary, for backward compatibility
+\subsectionAndLabel{How does the Checker Framework handle obsolete declaration annotations?}{faq-declaration-annotations-moved}
+
+When a declaration annotation is an alias for a type annotation, the
+Checker Framework may move the annotation before replacing it by the
+canonical version. (If the declaration annotation is in an \<org.checkerframework>
+package, it is not moved.)
+
+For example,
+
+\begin{Verbatim}
+  import android.support.annotation.NonNull;
+  ...
+  @NonNull Object [] returnsArray();
+\end{Verbatim}
+
+\noindent
+is treated as if the programmer had written
+
+\begin{Verbatim}
+  import org.checkerframework.checker.nullness.qual.NonNull;
+  ...
+  Object @NonNull [] returnsArray();
+\end{Verbatim}
+
+\noindent
+because Android's \<@NonNull> annotation is a declaration annotation, which
+is understood to apply to the top-level return type of the annotated method.
+
+The Checker Framework cannot always move an annotation written in a
```

Comment:
The problem isn't that the Checker Framework does move the annotation. The problem is when the user just changes the import from a declaration annotation to a type annotation.  I've added a new FAQ to try to address this.